src/M6Web/Bundle/ElasticsearchBundle/Connection/GuzzleConnectionDecorator.php

```diff
@@ -71,10 +75,34 @@ protected function dispatchEvent($method, $uri, $statusCode, $duration)
                 ->setUri($uri)
                 ->setMethod($method)
                 ->setStatusCode($statusCode)
-                ->setDuration($duration);
+                ->setDuration($duration)
+                ->setTook($took);
 
             $this->eventDispatcher->dispatch('m6web.elasticsearch', $event);
         }
     }
 
-}
\ No newline at end of file
+    /**
+     * Extract took from response
+     *
+     * @param array $response
+     *
+     * @return int|null
+     */
+    protected function extractTookFromResponse(array $response)
+    {
+        if (!isset($response['text'])) {
+
+            return null;
+        }
+
+        $result = json_decode($response['text'], true);
```

Comment:
I don't think `depth` can be used here. This parameter defined the max number of recursion. If this number is reached, `json_decode` returns `null`.

Let's try with a regex then.